component/src/main/java/io/siddhi/extension/io/file/listeners/FileCronExecutor.java

```diff
@@ -0,0 +1,212 @@
+                                   Map<String, String> properties, String fileURI,
+                                   FileSourceConfiguration fileSourceConfiguration, FileProcessor fileProcessor) {
+        vfsClientConnector.setMessageProcessor(fileProcessor);
+        BinaryCarbonMessage carbonMessage = new BinaryCarbonMessage(ByteBuffer.wrap(
+                fileURI.getBytes(StandardCharsets.UTF_8)), true);
+        try {
+            vfsClientConnector.send(carbonMessage, carbonCallback, properties);
+            try {
+                carbonCallback.waitTillDone(fileSourceConfiguration.getTimeout(), fileURI);
+            } catch (InterruptedException e) {
+                log.error(String.format("Failed to get callback from vfs-client  for file '%s'.",
+                        fileURI), e);
+            }
+            if (!fileSourceConfiguration.getActionAfterProcess().equalsIgnoreCase(Constants.KEEP)) {
+                reProcessFile(vfsClientConnector, carbonCallback, properties, fileURI, fileSourceConfiguration);
+            }
+        } catch (ClientConnectorException e) {
+            log.error(String.format("Failed to provide file '%s' for consuming.", fileURI), e);
+        }
+    }
+
+    public void reProcessFile(VFSClientConnector vfsClientConnector,
+                              VFSClientConnectorCallback vfsClientConnectorCallback,
+                              Map<String, String> properties, String fileUri,
+                              FileSourceConfiguration fileSourceConfiguration) {
+        String actionAfterProcess = fileSourceConfiguration.getActionAfterProcess();
+        properties.put(Constants.URI, fileUri);
+        properties.put(Constants.ACK_TIME_OUT, "1000");
+        BinaryCarbonMessage carbonMessage = new BinaryCarbonMessage(ByteBuffer.wrap(
+                fileUri.getBytes(StandardCharsets.UTF_8)), true);
+        String moveAfterProcess = fileSourceConfiguration.getMoveAfterProcess();
+        try {
+            if (fileSourceConfiguration.getActionAfterProcess() != null) {
+                properties.put(Constants.URI, fileUri);
+                properties.put(Constants.ACTION, actionAfterProcess);
+                File file = new File(fileSourceConfiguration.getUri());
+                if (fileSourceConfiguration.getMoveAfterProcess() != null) {
+                    if (file.isFile()) {
+                        properties.put(Constants.DESTINATION, fileSourceConfiguration.getMoveAfterProcess());
+                    } else {
+                        String destination = constructPath(moveAfterProcess, getFileName(fileUri,
+                                fileSourceConfiguration.getProtocolForMoveAfterProcess()));
+                        if (destination != null) {
+                            properties.put(Constants.DESTINATION, destination);
+                        }
+                    }
+                }
```

Comment:
Lets have another method to generate these properties as well. And use the same method [here](https://github.com/siddhi-io/siddhi-io-file/blob/master/component/src/main/java/io/siddhi/extension/io/file/listeners/FileSystemListener.java#L237) to remove duplicated code.

Reply:
fixed in 33a0d4ec677b82f844777943dec4b611303e759d